### ISSUE TITLE
Add explanation about Kleisli

### ISSUE THREAD

**Issue by wisn** [hemanth/functional-programming-jargon#161]
I found something like Kleisli composition (`>=>` in Haskell) in [Quora](https://www.quora.com/Haskell-programming-language-What-are-some-practical-uses-of-Kleisli-composition). I don't really understand the general of Kleisli. Based on Quora, Kleisli composition is a monadic function composition (CMIIW). How about this?

**Comment by jethrolarson** [comment]
Would be a good addition. Someone needs to break concept down to eli5 levels and then provide one or two examples

**Comment by MostAwesomeDude** [comment]
I can explain it, at least; I don't know of especially good examples.

Typically, we have functions, say of type X → Y. With a monad M, we might have actions ("impure" functions) in the monad, of type X → M(Y). Composition with functions happens to form a category: there is an identity arrow of type X → X (for any type X) which obeys the algebraic laws. Can we analogize this to monadic actions?

Yes, for *any* monad M. Kleisli composition lets us compose actions of type X → M(Y) and Y → M(Z) into actions of type X → M(Z), for any types X, Y, and Z. The identity arrow is provided by the unit of the monad, with type X → M(X).

I don't know when this is useful aside from "compiling to categories", a not-yet-common design for compiler pipelines. It probably could wait another few years.

**Comment by jethrolarson** [comment]
Let me try so break it down:

## Kleisli Composition

An operation for composing two monad-returning functions where they have compatible types.

```js
const foo = (a) => [a]
const bar = (a) => [a*2, a*3]

kleisliCompose(bar, foo)(1) // => [1, 2, 3]
```

**Comment by jethrolarson** [comment]
Admittedly array is a poor case and using Maybe would be more practical.
